packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -250,7 +250,7 @@ Future<XcodeBuildResult> buildXcodeProject({
     if (buildForDevice) {
       buildCommands.addAll(<String>['-sdk', 'iphoneos']);
     } else {
-      buildCommands.addAll(<String>['-sdk', 'iphonesimulator', '-arch', 'x86_64']);
+      buildCommands.addAll(<String>['-sdk', 'iphonesimulator']);
```

Comment:
\cc @gw280 FYI when you test your ARM framework you'll need to either comment out the two places I added `EXCLUDED_ARCHS[sdk=iphonesimulator*]=arm64 i386` or you can override the build settings in your test project.  Let me know if you need assistance.